### DISCUSSION TITLE
Wrong message displayed when a user registered

### DISCUSSION THREAD

**Discussion by luciolebrillante** [dani-garcia/vaultwarden#2898] [Q&A]
### Subject of the issue
When a new user registered, the green messages at the top right says in French that the user could now connect to the application whereas it is required to activate the account before.

### Deployment environment

* Vaultwarden version: 1.26 -  Version 2022.10.0 

* Install method:
* Manually installed

* Clients used: WEB
* 
* Other relevant details:

### Steps to reproduce
Register a new account and watch at the top right place

### Expected behavior
A message in French which says it is required to activate the account before logging to the application.

### Actual behavior
A message which says the user can now connect to the application.application.

**Comment by BlackDex** [comment]
Please provide the `Support String` which you can generate via the `/admin` interface when going to the `Diagnostics` page.

**Comment by luciolebrillante** [comment]
Hello @BlackDex,

Thank you for your help. You can find it below.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.26.0
* Web-vault version: v2022.10.0
* Running within Docker: false (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: PostgreSQL
* Database version: PostgreSQL 14.5 (Debian 14.5-2.pgdg110+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, ADMIN_TOKEN, SMTP_HOST, SMTP_PORT, SMTP_FROM, SMTP_FROM_NAME

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "/var/lib/vaultwarden/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "/var/lib/vaultwarden",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "**********://***********:********************************@*********:****/***********",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************.***************.**",
  "domain_origin": "*****://**************.***************.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "/var/lib/vaultwarden/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 24,
  "invitation_org_name": "XXX",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "/var/lib/vaultwarden/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "/var/lib/vaultwarden/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "***************.**,**********.***",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*****************************@***************.**",
  "smtp_from_name": "XXXX",
  "smtp_host": "********.**.***************.**.",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "off",
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "/var/lib/vaultwarden/templates",
  "tmp_folder": "/var/lib/vaultwarden/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "/usr/share/vaultwarden/web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Comment by BlackDex** [comment]
So, if I understand you correctly, when someone registers you want that account that it needs to verify it's email address first before they are able to login?

If that is the case, then you are missing the right config.
In that case you need to configure `signups_verify` to be `true`.

Also, all the messages sent by the server are in English. All messages shown by the web-vault it self will be shown in the language set by the user or defined by the browser.

**Comment by luciolebrillante** [comment]
Some things I observe :

* I enabled signups_verify in the main configuration file, it is the default behavior 
![Screenshot from 2022-11-07 12-53-07](https://user-images.githubusercontent.com/50231120/200303813-1625bd6e-484b-49df-bd10-094621c5eae0.png) but it is still written as false in the diagnostic panel -   "signups_verify": **false**, - and the user can still connect to the application without activiting his account.

* When I force it by true in the administration panel, it is now displayed as  "signups_verify": **true** in the diagnostic panel but I still get this message after I subscribed which means I can now connect to the application.![Screenshot from 2022-11-07 12-57-14](https://user-images.githubusercontent.com/50231120/200304843-9cd812f7-c417-4179-b467-3f5573c824ff.png)
However If I tried, I get this message in English (not in French) which said I can't. 
![Screenshot from 2022-11-07 12-58-01](https://user-images.githubusercontent.com/50231120/200304984-42348e1c-2c54-4619-bc59-5e6634d436ca.png)

Is there a way to choose another language for the messages sent by the server? I am ready to do the French translation if needed. I already made some fixes on the upstream Bitwarden app.

**Comment by BlackDex** [comment]
This specific feature is not part of Bitwarden, and a Vaultwarden only feature. This causes the register to work, and after that you need to verify the mail first. Because of this the message is in English.

So, if in the end you need to first verify as requested, and your only point is that the message is in English, then this is not an issue, but a works as intended.

**Comment by luciolebrillante** [comment]
Thank you for your answer about the localization message.

I think there are several issues here :

**The first one**
Whereas I have signups_verify : true in the configuration file, which should be the default behavior then,it is not detected by the application (diagnostic panel : signups_verify" signups_verify": false whereas it should be true and user can logging whereas they shouldn't.

**The second one:**
After I enabled signups_verify : true - After an user subscription, he receives a message telling him he can connect directly (the green message I sent) whereas he can't (mail activation required) (orange message I sent).

Behavior expected : received a message telling him he should enabled his account before trying to connect, for example, the same one in orange he will get after a connection try without activating his account.

**Reply by BlackDex** [reply]
As mentioned before, that isn't possible.
The force verify is a Vaultwarden specific feature. We can't return something else during the register flow, because that is not how Bitwarden works.

In the end, the feature works as intended. The language isn't selectable for the server side messages. Even though it does look a bit strange, it does work.

**Reply by luciolebrillante** [reply]
OK

About the second issue,  if I understand correctly, when an user receive a wrong information telling him he can connect whereas he can't. Is it supposed to work like that?

**Reply by BlackDex** [reply]
As mentioned before, we can't change the flow of the register or return something else then a `200`/`ok` response when someone registers.
It's either success or failed which the clients (like the web-vault) support. These messages are provided by the clients, and thus are provided in the language defined by the user or the browser.

The `signups_verify` feature is a Vaultwarden specific feature, not supported by the Bitwarden clients (like the web-vault), and thus we can only return a message when someone tries to login and check if the conditions are met or not.

Also, there already is a feature added in the `testing` tagged images (aka `main` branch) which will automatically verify users when they are invited via email (#2804). But if they just can register without an invite, then they still need to verify there email before they can login.

From the Bitwarden clients work-flow they should be able to just login, and the message is correct. But from the Vaultwarden side you are able to block this. Since that feature is a Vaultwarden specific (aka hacked) feature, we arn't able to provide a different message directly after the registration.

This is just how it is, we can't change this (unless Bitwarden is going to add a feature like this them self's of course).

**Reply by luciolebrillante** [reply]
Thank you for your completed answer! I understand better why it is not possible. I will be vigilant if there a new feature like that released in Bitwarden.
